src/rendering/SceneManager.cc

```diff
@@ -702,23 +702,52 @@ rendering::GeometryPtr SceneManager::LoadGeometry(const sdf::Geometry &_geom,
   }
   else if (_geom.Type() == sdf::GeometryType::MESH)
   {
-    auto fullPath = asFullPath(_geom.MeshShape()->Uri(),
-        _geom.MeshShape()->FilePath());
-    if (fullPath.empty())
+    common::MeshManager *meshManager =
+        common::MeshManager::Instance();
+    std::string meshUri;
+    rendering::MeshDescriptor descriptor;
+    if (meshManager->IsValidFilename(_geom.MeshShape()->Uri()))
```

Comment:
I see this logic is duplicated in Physics.cc. Can it be refactored?

Reply:
refactored into a helper function in Util class. [4f381ce](https://github.com/gazebosim/gz-sim/pull/2007/commits/4f381ce88d1f7a1c8d7718ad0bc6fdfbea97cec1)